crates/red_knot_python_semantic/src/types/generics.rs

```diff
@@ -100,7 +100,7 @@ impl<'db> GenericContext<'db> {
 }
 
 /// An assignment of a specific type to each type variable in a generic scope.
-#[salsa::tracked(debug)]
+#[salsa::interned(debug)]
```

Comment:
We currently have a problem on main, where we do not consider two instances of `C[int]` to be the same: https://playknot.ruff.rs/ee8f9cb9-a73f-480a-bb2c-83b6e1bb5e89, since we were comparing Salsa IDs in `is_assignable_to`. Thank you @dcreager for helping me in tracking (heh) this down.